src/extensions/ttt.py

```diff
@@ -0,0 +1,136 @@
+import discord
+
+import random
+
+
+players = [":x:", ":o:"]
+wordboard = [":one:", ":two:", ":three:",
+             ":four:", ":five:", ":six:",
+             ":seven:", ":eight:", ":nine:"]
+
+
+class tictactoe(commands.Cog):
+    def __init__(self, bot):
+        self.client = bot
+        self.__reset_board()
+        self.gameover = True
+
+    def __print_board(self):
+        line = ""
+        for i in range(3):
+            for j in range(3):
+                if self.__is_number(self.board[i*3 + j]):
+                    line += wordboard[i*3 + j]
+                else:
+                    line += self.board[i*3 + j]
+            line += "\n"
+        return line
+
+    def __reset_board(self):
+        self.board = [str(i+1) for i in range(9)]
+        self.currentplayer = random.choice(players)
+        self.gameover = False
+
+    def __check_win(self):
+        board = self.board
+        # diagonal:
+        if ((board[0] == board[4] == board[8]) or (board[2] == board[4] == board[6])) and not self.__is_number(board[4]):
+            return True
+        # check columns:
+        for i in range(3):
+            if board[i] == board[i+3] == board[i+6] and not self.__is_number(board[i]):
+                return True
+        # check rows:
+        for i in range(3):
+            if board[i*3] == board[i*3 + 1] == board[i*3 + 2] and not self.__is_number(board[i*3]):
+                return True
+        return False
+
+    def __check_tie(self):
+        for tile in self.board:
+            if self.__is_number(tile):
+                return False
+        return True
+
+    def __is_number(self, value):
+        try:
+            int(value)
+            return True
+        except Exception:
+            return False
+
+    def __select_tile(self, tile: int):
+        if tile >= 1 and tile <= 9 and self.__is_number(self.board[tile-1]):
+            self.board[tile-1] = self.currentplayer
+            return True
+        return False
+
+    def __switch_players(self):
+        self.currentplayer = players[1] if self.currentplayer == players[0] else players[0]
+
+    @commands.slash_command(
+        guild_ids=[1033025456242434089],
+        description="Play tic tac toe"
+    )
+    @discord.option(
+        "tile",
+        description="play the selected tile",
+        min_value=1,
+        max_value=9
+    )
+    @discord.option(
+        "reset",
+        description="reset the game board",
+        default=False
+        )
+    async def ttt(
+        self,
+        ctx: discord.ApplicationContext,
+        tile: int = discord.Option(
+            int, autocomplete=discord.utils.basic_autocomplete(
+            [i for i in range(9)])),
+        reset: bool = False,
+    ):
+        if reset or self.gameover:
+            self.__reset_board()
+            self.message = await ctx.respond(embed=self.__create_embed(f"Starting new game. \nIt's {self.currentplayer}'s turn", True))
+        elif self.__select_tile(tile):
```

Comment:
Could you turn the three different game state booleans into a single enum?
And could you turn this into a Match-Case Statement?

Reply:
Changed state check into enum in latest commit.